src/ethereum/spurious_dragon/state.py

```diff
@@ -306,7 +333,10 @@ def account_exists(state: State, address: Address) -> bool:
     account_exists : `bool`
         True if account exists in the state trie, False otherwise
     """
-    return get_account_optional(state, address) is not None
+    account = get_account_optional(state, address)
+    if account is None or account == EMPTY_ACCOUNT:
```

Comment:
Question: @ultratwo When will the account be None?

Reply:
Never mind, I figured it out.

Reply:
For anyone interested there are 2 possible scenarios:
1. Account does not exist because it has been destroyed. account is `None` in this case.
2. Account is empty when no value transfer or state has been set before.